interface/json/src/json.c

```diff
@@ -205,7 +205,13 @@ static cx_int16 cx_ser_item(cx_serializer s, cx_value *info, void *userData) {
 static cx_int16 cx_ser_complex(cx_serializer s, cx_value* v, void* userData) {
     cx_json_ser_t *data = userData;
     cx_type type = cx_valueType(v)->real;
-    if (!cx_ser_appendstr(data, "{")) {
+    cx_bool useCurlyBraces = FALSE;
+
+    if (type->kind == CX_COMPOSITE || cx_collection(type)->kind == CX_MAP) {
```

Comment:
This is dangerous - the type doesn't have to be a collection type in which case you're casting to a pointer of a different type. The correct comparison would be:

```C
if ((type->kind == CX_COMPOSITE) || ((type->kind == CX_COLLECTION) && (cx_collection(type)->kind == CX_MAP)) {
```